### PR TITLE
[SM-1444] Fix Windows GNU Builds

### PR DESCRIPTION
**Pull request by coltonhurst**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/SM-1444

## 📔 Objective

Windows GNU builds are currently broken due to a bug in [rustls-platform-verifier](https://github.com/rustls/rustls-platform-verifier) when LTO is turned on.

I have submitted a GH Issue for this here: https://github.com/rustls/rustls-platform-verifier/issues/141

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/build-rust-cross-platform.yml [hunk 1]
```diff
@@ -60,14 +60,23 @@ jobs:
       - name: Add build architecture
         run: rustup target add ${{ matrix.settings.target }}
 
+      # Build Rust for musl
       - name: Build Rust for - ${{ matrix.settings.target }}
```
Comment (michalchecinski) [inline .github/workflows/build-rust-cross-platform.yml:64]:
```suggestion
      - name: Build Rust for musl
```

File: .github/workflows/build-rust-cross-platform.yml [hunk 2]
```diff
@@ -60,14 +60,23 @@ jobs:
...
         if: ${{ contains(matrix.settings.target, 'musl') }}
         env:
           RUSTFLAGS: "-D warnings"
         run: cargo zigbuild -p bitwarden-c --target ${{ matrix.settings.target }} --release
 
+      # Build Rust for windows-gnu
       - name: Build Rust for - ${{ matrix.settings.target }}
```
Comment (michalchecinski) [inline .github/workflows/build-rust-cross-platform.yml:71]:
```suggestion
      - name: Build Rust for windows-gnu
```

**Review by michalchecinski — CHANGES_REQUESTED** [review]
I've added two renaming suggestions to reflect better which systems they should run for. This will result in better visibility of whether the appropriate build step has been run.

**Comment by coltonhurst** [comment]
@michalchecinski I think it might be beneficial to leave the explicit targets as is?

For `musl`, it could be two different targets, `x86_64-unknown-linux-musl` or `aarch64-unknown-linux-musl`. If we just make it `Build rust for musl` it won't be immediately obvious which architecture it is.

**Review by michalchecinski — APPROVED** [review]

**Pull request merged**